### PR TITLE
Fixes #12472 - compute resource specific details in host help

### PR DESCRIPTION
**Pull request by tstrachota**
Requires https://github.com/theforeman/hammer-cli/pull/222
- host create and update commands provide interface for adding compute resource specific information into help
- new module `ComputeResources` for compute resource specific behaviour, the file structure counts with adding more extension types in future (http://projects.theforeman.org/issues/4593, http://projects.theforeman.org/issues/10311, http://projects.theforeman.org/issues/3651, ...)
- host extension modules moved to module `Hosts`

### CODE REVIEW

**Comment by mbacovsky** [comment]
[test]

File: lib/hammer_cli_foreman/hosts/common_update_help.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+module HammerCLIForeman
+  module Hosts
+    module CommonUpdateHelp
+      def self.included(base)
+        base.extend_help do |h|
+          h.section 'Available keys for --interface' do |h|
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hosts/common_update_help.rb]: Translation is missing

File: lib/hammer_cli_foreman/hosts/common_update_help.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+              'name',
+              'subnet_id',
+              'domain_id',
+              'identifier',
+              ['managed',   'true/false'],
+              ['primary',   _('%{value}, each managed hosts needs to have one primary interface.') % {:value => 'true/false'}],
+              ['provision', 'true/false'],
+              ['virtual',   'true/false']
+            ])
+            h.section 'For virtual=true' do |h|
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hosts/common_update_help.rb]: Translation is missing

File: lib/hammer_cli_foreman/hosts/common_update_help.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+              h.list([
+                ['tag',         _('VLAN tag, this attribute has precedence over the subnet VLAN ID. Only for virtual interfaces.')],
+                ['attached_to', _('Identifier of the interface to which this interface belongs, e.g. eth1.')]
+              ])
+            end
+            h.section 'For type=bond' do |h|
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hosts/common_update_help.rb]: Translation is missing

File: lib/hammer_cli_foreman/hosts/common_update_help.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+              h.list([
+                ['mode',             _('Possible values: %s') % 'balance-rr, active-backup, balance-xor, broadcast, 802.3ad, balance-tlb, balance-alb'],
+                ['attached_devices', _('Identifiers of slave interfaces, e.g. [eth1,eth2]')],
+                'bond_options'
+              ])
+            end
+            h.section 'For type=bmc' do |h|
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hosts/common_update_help.rb]: Translation is missing

File: lib/hammer_cli_foreman/hosts/common_update_help.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              h.list([
+                ['provider', _('always IPMI')],
+                'username',
+                'password'
+              ])
+            end
+          end
+
+          h.section 'Provider specific options' do |h|
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hosts/common_update_help.rb]: Translation is missing

**Comment by tstrachota** [comment]
@mbacovsky updated

**Comment by mbacovsky** [comment]
:+1: Looks ggot to merge now, thanks @tstrachota! :poodle:

**Pull request merged**